src/charm.py

```diff
@@ -195,6 +202,20 @@ def __init__(self, *args):
         # Action handlers
         observe(self.on.show_proxied_endpoints_action, self._on_show_proxied_endpoints)
 
+    def _on_cert_transfer_relation_joined(self, event: RelationJoinedEvent):
+        if self.cert.enabled:
+            self.cert_transfer.set_certificate(
+                self.cert.cert,
+                self.cert.ca,
+                self.cert.chain,
+                relation_id=event.relation.id,
+            )
+
+    def _on_cert_transfer_relation_departed(self, event: RelationDepartedEvent):
+        self.cert_transfer.remove_certificate(
+            relation_id=event.relation.id,
+        )
```

Comment:
Do we need this for the provider side?
1. If traefik is going away, we don't care because the other side needs to take care of it.
2. If the other side is going away, then juju will automatically cleanup the relation anyway.

Reply:
the library docs recommended to do this, I followed blindly. But I get your point and I think you're right. Should we ask the library owners for clarity, or just do what we think is right?

Reply:
Yes :)